langchain/src/vectorstores/qdrant.ts

```diff
@@ -0,0 +1,116 @@
+import { v4 as uuid } from "uuid";
+import type { QdrantClient } from "@qdrant/js-client-rest";
+
+import { Embeddings } from "../embeddings/base.js";
+import { VectorStore } from "./base.js";
+import { Document } from "../document.js";
+
+interface QdrantSearchResponse {
+  id: number;
+  score: number;
+  payload: {
+    content: string;
+    metadata: object;
+  };
+}
+
+export interface QdrantLibArgs {
+  client: QdrantClient;
+  collectionName?: string;
+}
+
+export class QdrantVectorStore extends VectorStore {
+  client: QdrantClient;
+
+  collectionName: string;
+
+  constructor(embeddings: Embeddings, args: QdrantLibArgs) {
+    super(embeddings, args);
+    this.client = args.client;
```

Comment:
Instead of taking a client as an arg here, can we initialize the client internally from a config object like this?

https://github.com/hwchase17/langchainjs/pull/1271/files#diff-87a2d06951a95c0abda758677c0617d601b183627559a1cb331cf184e91fe56bR37

Reply:
well, I support both taking a exist client as an arg and initializing a client internal, and will verify whether the collection exists.